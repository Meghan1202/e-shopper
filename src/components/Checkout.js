import React from 'react';

const Checkout = () => (
  <div>Thank you for shoping with us!</div>
);

export default Checkout;
