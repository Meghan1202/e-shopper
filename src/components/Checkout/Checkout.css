.Checkout__Container{
    display: flex;
    flex-wrap: wrap;
    width: max-content;
    height: max-content;
}