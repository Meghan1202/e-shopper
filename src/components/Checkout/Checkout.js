import React from 'react';
import './Checkout.css';
import { Formik, Form, Field } from 'formik';

// function validateEmail(value) {
//   let error;
//   if (!value) {
//     error = 'Required';
//   } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)) {
//     error = 'Invalid email address';
//   }
//   return error;
// }

function validateName(value) {
  console.log({ value });
  let error;
  if (!value) {
    error = 'Required';
  }
  return error;
}

function validatePhone(value) {
  let error;
  if (!value) {
    error = 'Required';
  } else if (!/^[789]\d{9}$/i.test(value)) {
    error = 'Invalid Phone Number';
  }
  return error;
}

// convert to functional component and use hook
class Checkout extends React.Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      mobile: '',
      checkoutMessage: '',
    };
  }

  checkout = (event) => {
    this.handleCheck(event);
    console.log(event);
    this.setState({ checkoutMessage: 'Thank you for shopping with us!' });
  }

  handleCheck= (event) => {
    if (event.target.name === 'name') {
      this.setState((prevState) => ({
        ...prevState,
        [event.target.name]: event.target.value,
      }));
    }
    if (event.target.name === 'email') {
      this.setState((prevState) => ({
        ...prevState,
        [event.target.name]: event.target.value,
      }));
    }
    if (event.target.name === 'mobile') {
      this.setState((prevState) => ({
        ...prevState,
        [event.target.name]: event.target.value,
      }));
    }
  }

  render() {
    const {
      checkoutMessage,
    } = this.state;
    return (
      <>
        <Formik
          initialValues={{
            name: '',
            email: '',
            mobile: '',
          }}
          onSubmit={(values, actions) => {
            setTimeout(() => {
              alert(JSON.stringify(values, null, 2));

              actions.setSubmitting(false);
            }, 1000);
          }}
        >
          {({ errors, touched }) => (
            <Form>
              <span>First name:</span>
              <br />
              <Field
                name="name"
                // onChange={this.handleCheck}
                validate={validateName}
                // value={name}
              />
              {errors.name && touched.name && <div>{errors.name}</div>}
              <br />
              <span>Email:</span>
              <br />
              <Field
                name="email"
                // onChange={this.handleCheck}
                type="email"
                // validate={validateEmail}
                // value={email}
              />
              {errors.email && touched.email && <div>{errors.email}</div>}
              <br />
              <span>Mobile Number:</span>
              <br />
              <Field
                name="mobile"
              // onChange={this.handleCheck}
                validate={validatePhone}
                // typ
                // value={mobile}
              />
              {errors.mobile && touched.mobile && <div>{errors.mobile}</div>}
              <br />
              <button type="button" onClick={this.checkout}>submit</button>
            </Form>
          )}
        </Formik>
        <p>{checkoutMessage}</p>
      </>
    );
  }
}
export default Checkout;
