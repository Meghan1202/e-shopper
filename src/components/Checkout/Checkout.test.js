import axios from 'axios';
import React from 'react';
import { render, screen } from '@testing-library/react';
import Checkout, { postCartData } from './Checkout';

describe(Checkout.name, () => {
  beforeEach(() => {
    const mockResponse = {
      data: {
        items: [
          {
            id: 4,
            name: 'duster',
            price: 80,
            count: 1,
            category: 'Household Items',
          },
        ],
        id: 60,
        date: 1615541692008,
      },
    };
    jest.spyOn(axios, 'post').mockResolvedValueOnce(mockResponse).mockResolvedValue(null);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('Should mock the post call', async () => {
    const mockResponse = {
      data: {
        items: [
          {
            id: 4,
            name: 'duster',
            price: 80,
            count: 1,
            category: 'Household Items',
          },
        ],
        id: 60,
        date: 1615541692008,
      },
    };
    const mockCartData = [];
    const mockUpdateAllOrders = jest.fn();
    const mockPastOrderData = [];
    const mockResetCart = jest.fn();
    const mockResetProductCache = jest.fn();
    const spyOnAxios = jest.spyOn(axios, 'post').mockResolvedValueOnce(mockResponse);
    await postCartData(mockCartData, mockUpdateAllOrders, mockPastOrderData,
      mockResetCart, mockResetProductCache);
    expect(spyOnAxios).toHaveBeenCalledTimes(1);
  });
  test('should match the snapshot', () => {
    const mockcheckoutItems = jest.fn();
    const mockupdateAllOrders = jest.fn();
    const mockallorders = [];
    const mockResetCart = jest.fn();
    const mockResetProductCache = jest.fn();
    const { container } = render(<Checkout
      checkoutItems={mockcheckoutItems}
      updateAllOrders={mockupdateAllOrders}
      allOrders={mockallorders}
      resetCart={mockResetCart}
      resetProductCache={mockResetProductCache}
    />);
    expect(container).toMatchSnapshot();
  });
  test('should render the submit button', () => {
    const mockcheckoutItems = jest.fn();
    const mockupdateAllOrders = jest.fn();
    const mockallorders = [];
    const mockResetCart = jest.fn();
    const mockResetProductCache = jest.fn();
    render(<Checkout
      checkoutItems={mockcheckoutItems}
      updateAllOrders={mockupdateAllOrders}
      allOrders={mockallorders}
      resetCart={mockResetCart}
      resetProductCache={mockResetProductCache}
    />);
    screen.getByTestId('Submit-button');
  });
});
