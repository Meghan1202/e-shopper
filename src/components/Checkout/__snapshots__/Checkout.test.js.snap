// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout should match the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Checkout
    </h1>
    <p>
      We'll require your contact information to send the items in your cart
    </p>
    <form
      class="checkoutForm"
    >
      <label
        for="firstName"
      >
        First Name:
        <input
          name="firstName"
          placeholder="John"
          type="text"
        />
      </label>
      <label
        for="lastName"
      >
        Your name:
        <input
          name="lastName"
          placeholder="Doe"
          type="text"
        />
      </label>
      <label
        for="email"
      >
        Your email:
        <input
          name="email"
          placeholder="john_doe@example.com"
          type="text"
        />
      </label>
      <label
        for="phoneNumber"
      >
        Your email:
        <input
          name="phoneNumber"
          placeholder="987654321"
          type="text"
        />
      </label>
      <button
        class="placeOrder primary"
        data-testid="Submit-button"
        type="submit"
      >
        Place order
      </button>
    </form>
  </div>
</div>
`;
